libs/core/langchain_core/language_models/chat_models.py

```diff
@@ -103,8 +104,16 @@ async def agenerate_from_stream(
 class BaseChatModel(BaseLanguageModel[BaseMessage], ABC):
     """Base class for Chat models."""
 
-    cache: Optional[bool] = None
-    """Whether to cache the response."""
+    cache: Optional[bool | BaseCache] = None
```

Comment:
@alhridoy I made a few changes in this file. We should document the property itself.

We can ignore examples for now. (It would be nice to get documentation in the main python site, but we can do that afterwards.)

The main change was to streamline the caching path / non caching code paths, so there's only one path for caching related logic independently of whether a local cache is used or a global one